plugins/woocommerce/src/Blocks/BlockTemplatesRegistry.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Utils\BlockTemplateUtils;
+use Automattic\WooCommerce\Blocks\Templates\AbstractTemplate;
+use Automattic\WooCommerce\Blocks\Templates\AbstractTemplatePart;
+use Automattic\WooCommerce\Blocks\Templates\MiniCartTemplate;
+use Automattic\WooCommerce\Blocks\Templates\CartTemplate;
+use Automattic\WooCommerce\Blocks\Templates\CheckoutTemplate;
+use Automattic\WooCommerce\Blocks\Templates\CheckoutHeaderTemplate;
+use Automattic\WooCommerce\Blocks\Templates\OrderConfirmationTemplate;
+use Automattic\WooCommerce\Blocks\Templates\ProductAttributeTemplate;
+use Automattic\WooCommerce\Blocks\Templates\ProductCatalogTemplate;
+use Automattic\WooCommerce\Blocks\Templates\ProductCategoryTemplate;
+use Automattic\WooCommerce\Blocks\Templates\ProductTagTemplate;
+use Automattic\WooCommerce\Blocks\Templates\ProductSearchResultsTemplate;
+use Automattic\WooCommerce\Blocks\Templates\SingleProductTemplate;
+
+/**
+ * BlockTemplatesRegistry class.
+ *
+ * @internal
+ */
+class BlockTemplatesRegistry {
+
+	/**
+	 * The array of registered templates.
+	 *
+	 * @var AbstractTemplate[]|AbstractTemplatePart[]
+	 */
+	private $templates = array();
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
```

Comment:
I think we should invoke the init method in the Bootstrap class. Doing so will make it a bit easier for testing because nothing is executed when a new instance is initialized.

Reply:
Makes sense. Done in 713c43afdbbc067c0e639e5022afa84925d982bf.